sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -614,7 +628,10 @@ public virtual async Task StartProcessingAsync(CancellationToken cancellationTok
         ///
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the stop operation.  If the operation is successfully canceled, the <see cref="EventProcessorClient" /> will keep running.</param>
         ///
-        public virtual async Task StopProcessingAsync(CancellationToken cancellationToken = default)
+        public virtual void StopProcessing(CancellationToken cancellationToken = default)
+            => StopProcessingImplAsync(false, cancellationToken).EnsureCompleted();
+
+        private async Task StopProcessingImplAsync(bool async, CancellationToken cancellationToken)
```

Comment:
async is not used.

Reply:
I was just about to mention the same.  This method appears to do the same exact thing as the pre-refactor version, though it takes this parameter.

Reply:
> async is not used.

That would be in part 2 with AZC0106-AZC0111 enabled. I can enable them in this PR.

Reply:
Added a sample change to `StopProcessingInternalAsync` to show the purpose of the parameter. It is not a full change cause it looks like it will be complicated in this case.

Reply:
Thanks.   I get the gist now.  We're basically migrating from having a top level that hides the async flow we use all the way down and, instead, adding another internal level that manages the async away.  What I'm still not sure about is what our motivation is for doing so.

Reply:
I've created an [issue](https://github.com/Azure/azure-sdk-for-net/issues/10083) with explanation.

Reply:
Thank you.  That was incredibly helpful.  Understanding the full context definitely helps to drive home the need and ensure that we're doing the right thing in any future enhancements.

//cc: @kinelski _(please read the issue, as we'll need to keep this in mind going forward)_